### PR TITLE
[FLINK-10279] [documentation] Make jython limitations more obvious in documentation.

### PR DESCRIPTION
**Pull request by tweise**
Purpose of this change is to better highlight the potential restrictions of the Jython based streaming API to avoid user confusion.

### CODE REVIEW

**Comment by tweise** [comment]
CC: @StephanEwen @aljoscha @zentol

File: docs/dev/stream/python.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+*Note: This API is based on Jython, which is not a full Python replacement and may restrict
```
Comment (zentol) [inline docs/dev/stream/python.md]: You can actually create neat looking `attention` boxes, like so `<span class="label label-danger">Attention</span>`, as seen here: https://ci.apache.org/projects/flink/flink-docs-master/dev/event_timestamps_watermarks.html#assigning-timestamps

File: docs/dev/stream/python.md [hunk 2]
```diff
@@ -50,7 +53,12 @@ existing Java streaming APIs.
 There are two main constraints for using Jython:
 
 * The latest Python supported version is 2.7
-* It is not straightforward to use Python C extensions
+* It is not straightforward to use Python C extensions, which may prevent reuse of some libraries
+
+(For more information please see <https://wiki.python.org/jython/JythonFaq/GeneralInfo>.)
+
+One possible alternative for streaming that allows for native Python execution would be the Apache Beam
```
Comment (zentol) [inline docs/dev/stream/python.md]: add a link to the Apache Beam project, or maybe even a more specific python-related page.

**Comment by tweise** [comment]
@zentol @mxm thanks, I incorporated your suggestions. PTAL

File: docs/dev/stream/python.md [hunk 3]
```diff
@@ -50,7 +54,12 @@ existing Java streaming APIs.
 There are two main constraints for using Jython:
 
 * The latest Python supported version is 2.7
-* It is not straightforward to use Python C extensions
+* It is not straightforward to use Python C extensions, which may prevent reuse of some libraries
```
Comment (mxm) [inline docs/dev/stream/python.md]: Would rephrase `reuse` to `use`.
